### PR TITLE
[script] [common] Add method to parse item name as text into DRC::Item instance

### PR DESCRIPTION
**Pull request by KatoakDR**
### Background
* My plan is to use this method in both `common-items` and `equipmanager` to "robustify" their methods to handle arguments of either an Item instance or text to enable the internal APIs to standardize on a canonical model (the Item class) when operating on items.
* It also provides an easy way to convert what's in your hand into an Item class instance (e.g. `DRC::Item.from_text(DRC.right_hand)` without the calling code needing to split the string and determine if need to set the `adjective` and/or `name` properties.
* This is also motivated by https://github.com/rpherbig/dr-scripts/pull/4980, which had to revert a use of `DRCI.in_hands?(item.short_name)` because of how that method was implemented trying to parse the string into an Item class instance. Words that were joined by a period ("blue.sword") were treated as one word and didn't split. This proposed method will alleviate mishaps like that in the future.

### Changes
* Add static method `from_text` to parse an item's name (as shown when held in your hand) into a `DRC::Item` class instance

I have planned subsequent PRs that will make use of this method.

## Tests

<details>
<summary>nil or blank string</summary>

```
> ,e echo DRC::Item.from_text(nil).inspect
> ,e echo DRC::Item.from_text('').inspect
> ,e echo DRC::Item.from_text('   ').inspect

--- Lich: exec1 active.

[exec1: nil]

--- Lich: exec1 has exited.
```
</details>

<details>
<summary>sword</summary>

```
> ,e echo DRC::Item.from_text('sword').inspect

--- Lich: exec1 active.

[exec1: #<DRC::Item:0x10490c40 @name="sword", @leather=nil, @worn=false, @hinders_lockpicking=nil, @container=nil, @swappable=false, @tie_to=nil, @adjective=nil, @bound=false, @wield=false, @transforms_to=nil, @transform_verb=nil, @transform_text=nil, @lodges=true, @skip_repair=false, @ranged=false, @needs_unloading=false>]

--- Lich: exec1 has exited.
```
</details>

<details>
<summary>steel sword</summary>

```
> ,e echo DRC::Item.from_text('steel sword').inspect

--- Lich: exec1 active.

[exec1: #<DRC::Item:0x12e9ea70 @name="sword", @leather=nil, @worn=false, @hinders_lockpicking=nil, @container=nil, @swappable=false, @tie_to=nil, @adjective="steel", @bound=false, @wield=false, @transforms_to=nil, @transform_verb=nil, @transform_text=nil, @lodges=true, @skip_repair=false, @ranged=false, @needs_unloading=false>]

--- Lich: exec1 has exited.
```
</details>

<details>
<summary>rusty copper sword</summary>

```
> ,e echo DRC::Item.from_text('rusty copper sword').inspect

--- Lich: exec1 active.

[exec1: #<DRC::Item:0x12e9ea70 @name="sword", @leather=nil, @worn=false, @hinders_lockpicking=nil, @container=nil, @swappable=false, @tie_to=nil, @adjective="rusty", @bound=false, @wield=false, @transforms_to=nil, @transform_verb=nil, @transform_text=nil, @lodges=true, @skip_repair=false, @ranged=false, @needs_unloading=false>]

--- Lich: exec1 has exited.
```

### CODE REVIEW

**Comment by wstampley** [comment]
Unrelated to your PR, but it is super weird to me that the 'item' class is not in 'common-items'.

File: common.lic [hunk 1]
```diff
@@ -349,6 +349,23 @@ module DRC
     def short_regex
       @adjective ? /\b#{adjective}.*\b#{@name}/i : /\b#{@name}/i
     end
+
+    # Convenience method to parse the text of an item as shown when in your hands, with or without an adjective,
+    # into an Item class instance. Originally designed to support DRCI and equipmanager methods.
+    def self.from_text(text)
```
Comment (wstampley) [inline common.lic:355]: I guess I see why you're doing this, but it's a pretty weakly built item (missing 95% of its properties, etc.).

Comment (KatoakDR) [reply common.lic:355]: Yeah, the goal is to take the text of an item, get it into the Item class in a consistent way **then** make use of the Item class' `short_regex` and `short_name` properties.

In short, have common-items and equipmanager standardize onto a canonical internal data model for items and the regex pattern to match them.

Outside of equipmanager's use, yeah, the additional properties just aren't being used.

We _could_ in good object-oriented or composition design have a simpler class that's for the item adj/name and then the class we have today would extend or use that class.

**Comment by KatoakDR** [comment]
> Unrelated to your PR, but it is super weird to me that the 'item' class is not in 'common-items'.

Agree, and I haven't explored why it's in common vs. common-items. I don't _think_ it has to be in common, and if not then a future PR could move it.

**Review by wstampley — APPROVED** [review]

**Comment by KatoakDR** [comment]
@rpherbig this is ready for your final review, thanks!

**Comment by rpherbig** [comment]
The code changes look good ✔️ 

Can you resolve the merge conflict?

**Comment by KatoakDR** [comment]
> The code changes look good ✔️
> 
> Can you resolve the merge conflict?

Conflict resolved, thanks

**Pull request merged**